src/ExExpression.elm

```diff
@@ -272,13 +273,48 @@ flattenTypeApplication application =
             [ other ]
 
 
+{-| Returns uncurryied function application if arguments length is matching definition arity
+otherwise returns curried version
+-}
+functionApplication : Context -> Expression -> Expression -> String
+functionApplication c left right =
+    let
+        reduceArgs c args =
+            args |> List.map (elixirE c) |> String.join ", "
+    in
+        (case applicationToList (Application left right) of
+            (Variable [ fn ]) :: args ->
+                if hasMatchingArity c c.mod fn args then
+                    [ toSnakeCase True fn, "(", reduceArgs c args, ")" ]
+                else
+                    [ elixirE c left, ".(", elixirE c right, ")" ]
+
+            (Access (Variable modules) [ fn ]) :: args ->
+                let
+                    mod =
+                        modulePath modules
+
+                    fnName =
+                        (toSnakeCase True fn)
+                in
+                    if hasMatchingArity c mod fn args then
+                        [ mod, ".", fnName, "(", reduceArgs c args, ")" ]
+                    else
+                        [ mod, ".", fnName, "().(", args |> List.map (elixirE c) |> String.join ").(", ")" ]
+
+            _ ->
+                [ elixirE c left, ".(", elixirE c right, ")" ]
+        )
+            |> String.join ""
```

Comment:
Not sure what would be an other way to solve it, but I'm not a fan of this solution